### PR TITLE
feat!: Change enterprise runner to also use ListRunnersOptions

### PR DESCRIPTION
**Pull request by ponkio-o**
Fixes: #3166.

BREAKING-CHANGE: This changes `ListOptions` to  `ListRunnersOptions` in `EnterpriseService.ListRunners`.

I fixed a bug related ListRunners API. The `ListRunnersOptions` was not being applied to Enterprise Runner.
follow up: https://github.com/google/go-github/pull/3094

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @ponkio-o !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gmlewis** [comment]
@fentas - perhaps you are available to review this?

**Review by fentas — APPROVED** [review]
lgtm

**Comment by gmlewis** [comment]
Thank you, @fentas ! Merging.

**Pull request merged**
